compare_solver_exercise.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## ------- import packages -------
+import networkx as nx
+import dimod
+# TODO:  Add code here to import your sampler
+
+# TODO:  Add code here to import your Traveling Salesman QUBO generator
+
+
+## ------- provided distance matrix -------
+distance_matrix = [
```

Comment:
If we're eventually going to make a graph, why not just make the graph directly here? E.g.

```
G = nx.Graph()
G.add_weighed_edges_from([
    (0, 1, 2230),
    (0, 2, 1631),
    ...
])
```
I found it confusing that you make a matrix here but `get_qubo` takes a graph.

Reply:
The idea was that the user would call the appropriate method in dwave_networkx to create the QUBO. I wanted to provide the graph, but ask the user to find the method in dwave_networkx.

Reply:
I think Alex is asking about the content outside of the ``get_qubo`` function.

Current workflow in your program is this:
Build distance matrix -> Create graph -> Add edges according to distance matrix -> pass graph to get_qubo function.

He is suggesting this:
Create graph -> add weighted edges to graph directly (no distance matrix data structure required) -> pass graph to get_qubo function.

Reply:
Yes, I am suggesting that we minimize the number of steps taken outside of the parts the user is responsible for.

Reply:
I've updated this. Please have a look.